### ISSUE TITLE
Inkplate 10 not charging

### ISSUE THREAD

**Issue by werdnum** [SolderedElectronics/Inkplate-Arduino-library#188]
Hi folks,

I have one of the Inkplate 10s from the original crowd funding round, I guess I received it late 2021.

I'm finding that the battery does not seem to charge when the unit is plugged into the wall (I'm mostly testing using a USB C-C cable attached either to my laptop or to a PD charger. The device will run, but the battery voltage does not increase, and one time I emptied the battery completely and then plugged the device in overnight, and the battery was still empty afterwards.

The battery in question seems to be chargeable by an external device (DFRobot DFR0446) but not by the Inkplate itself. I've tried various variations of having the Inkplate off or on.

Do you have any particular troubleshooting steps to try to figure out why this isn't working?

**Comment by misel228** [comment]
If I understand it correctly, you've tested only PD aware power supplies.

I'm not sure about the Inkplate 10, but I know for a fact that my Inkplate 6 COLOR doesn't have the required resistors for these smart power supplies to detect a 5V sink. In this case the PD standard requires the power supply to shut off and so it doesn't charge.

I have to charge my Inkplate with a dumb charger and a USB-A to USB-C cable.

Soldered Electronics is in good company with this fault they're certainly not the first ones to do that and I fear they won't be the last. This is especially sad since this topic has already entered the mainstream a few years ago when the first iteration of the Raspberry Pi 4 (the first with a USB-C port) had the same error.

Here is just one article about the issue with a few references https://www.tomshardware.com/news/raspberry-pi-4-usb-c-update

**Comment by werdnum** [comment]
> If I understand it correctly, you've tested only PD aware power supplies.
> 
> I'm not sure about the Inkplate 10, but I know for a fact that my Inkplate 6 COLOR doesn't have the required resistors for these smart power supplies to detect a 5V sink. In this case the PD standard requires the power supply to shut off and so it doesn't charge.
> 
> I have to charge my Inkplate with a dumb charger and a USB-A to USB-C cable.
> 
> Soldered Electronics is in good company with this fault they're certainly not the first ones to do that and I fear they won't be the last. This is especially sad since this topic has already entered the mainstream a few years ago when the first iteration of the Raspberry Pi 4 (the first with a USB-C port) had the same error.
> 
> Here is just one article about the issue with a few references https://www.tomshardware.com/news/raspberry-pi-4-usb-c-update

Hmm, thanks for this. Let me try an A-to-C cable on a USB-A port and get back to you.

**Comment by werdnum** [comment]
I tried a regular A-to-C cable. I found that the Inkplate will run off it, but it only seems to charge up to 3.14V or thereabouts before stopping.

See attached screenshot of the voltage over time according to Home Assistant.

![Screenshot of battery voltage over time as measured by ESPHome and stored in Home Assistant. The graph shows the voltage fairly stable over time at around 3.6V until it drops precipitously to 3V and disappears. A little later, the graph rises rapidly from 3V up to around 3.1V, but it seems to top out a little above 3.1V](https://user-images.githubusercontent.com/271070/224609093-26dce280-010d-4e4e-a6dd-fba3710c4348.png)

**Comment by ifadiga** [comment]
Hi, 
This seems like your Inkplate has a faulty charger. Can you connect Inkpalte to USB and measure the voltage on battery pins without a connected battery?  The voltage needs to be between 4.1 and 4.2V.

**Comment by BornaBiro** [comment]
@misel228 There are separate resistors on the CC1 and CC2 on the USB-C connector as seen on the [schematic](https://github.com/SolderedElectronics/Inkplate-10-hardware/blob/main/Schematics%2C%20Gerber%2C%20BOM/v1.0/Inkplate10%20schematics%20v1.0.pdf).
![image](https://user-images.githubusercontent.com/43830307/224645195-526bf42c-d62a-48ec-b098-6e87aaefcfb9.png)
I tried multiple Inkplate 10 with multiple USB-C to USC-C cables and it powers up and changes the battery just fine.

@werdnum
Does it run of the battery? And what kind of battery did you use? There are some batteries that have reversed polarity, so by connecting this battery to the JST connector, you could easily damage the charger IC. If this is not the case, then is definitely a faulty MCP73831T charger IC.

**Comment by werdnum** [comment]
> Does it run of the battery? And what kind of battery did you use? There are some batteries that have reversed polarity, so by connecting this battery to the JST connector, you could easily damage the charger IC. If this is not the case, then is definitely a faulty MCP73831T charger IC.

Yes, it does run off the battery. It just doesn't seem to charge.

It's this kind of battery: https://core-electronics.com.au/polymer-lithium-ion-battery-1000mah-38458.html. I haven't used any other kind of battery.

> This seems like your Inkplate has a faulty charger. Can you connect Inkpalte to USB and measure the voltage on battery pins without a connected battery? The voltage needs to be between 4.1 and 4.2V.

Let me pull it apart and give this a go and get back to you...

**Comment by werdnum** [comment]
Got the case off and a multimeter out.

- multimeter agrees that battery voltage is 3.11V.
- When disconnecting the battery, the voltage across the VBatt terminals starts at 2.6V and drops down to zero continuously as the multimeter is connected, over the course of a minute or so - so perhaps a capacitor is being drained?

Attached a couple of quick photos in case something jumps out at you.

![9E324282-BD3C-40E8-86D9-A8D5E6681384](https://user-images.githubusercontent.com/271070/224682211-81ff3b5f-e5d1-401e-a04f-505199d097ce.jpeg)
![6674751D-6831-49E4-B1B8-E3F1FE99B3AE](https://user-images.githubusercontent.com/271070/224682146-bb68534d-6e48-41a5-b26f-4397cfb6cb6e.jpeg)
![8E97CBA8-D317-4D98-BC08-99C63A95CB61](https://user-images.githubusercontent.com/271070/224682188-4f9d7d81-fdbc-46c3-b0a9-cd8b05361936.jpeg)
![724517D4-C3FD-4751-BECA-6142570907D3](https://user-images.githubusercontent.com/271070/224682194-49f23b6c-c579-4880-aa12-059805f9a3cb.jpeg)

**Comment by ifadiga** [comment]
Hi, 
I see in the picture that you don't have a charger IC on the board.

Please contact us at contact@soldered.com, send photos, and write that you don't have a charger IC on your Inkplate board and you will get a replacement board.

**Comment by Stanko** [comment]
Sorry to bump this issue, but I have the same problem.

To be fair I connected the battery before I swapped it's polarity. When I swapped it, Inkplate works fine on battery, but the charge LED doesn't come on when it is plugged in.

If I managed to damage something, is it possible to fix it.

Here are two photos in a case they help:
|Battery power|Plugged in|
|-|-|
|![IMG_2718](https://user-images.githubusercontent.com/776788/230093322-42c4ae22-ae59-4fd9-b30a-c4925de27aee.jpg)|![IMG_2719](https://user-images.githubusercontent.com/776788/230093354-ea871669-3657-4a14-98b0-c12967c0334d.jpg)|

It can be hard to see, but on both images blue power LED is on.

Thank you!

**Comment by ifadiga** [comment]
Hi,
If you have a problem with charging you probably damaged the charger IC. That is a little IC near the CHRG LED. You can change it but be careful because if you heat PCB you can damage the e-paper panel. We use MCP73831T for the charger.

**Comment by Stanko** [comment]
Thank you! I'll try that.
